src/database/sql_database.cc

```diff
@@ -2358,7 +2355,8 @@ std::vector<std::shared_ptr<CdsResource>> SQLDatabase::retrieveResourcesForObjec
     while ((row = res->nextRow())) {
         auto resource = std::make_shared<CdsResource>(std::stoi(getCol(row, ResourceCol::handlerType)));
         resource->decode(getCol(row, ResourceCol::options), getCol(row, ResourceCol::parameters));
-        resource->setResId(resources.size());
+        int res_id = row->col_int(to_underlying(ResourceCol::res_id), 0);
+        resource->setResId(res_id);
```

Comment:
The idea was to ensure that there is no gap in ids when loading - the database may have been changed.

Reply:
In that case, I can revert that change